### PR TITLE
fix: js and json lang quotes escape

### PR DESCRIPTION
**Pull request by alexgarel**
Translations coming from crowdin might contain single quotes or double quotes. So whenever we use lang in a json or javascript string we should escape quotes.

I created edq and esq (Escape Single/Double Quotes) function for that (1st commit) and applied it in a lot of place (2nd commit). This should be the pattern to use from now on. 

fixes: 
- #9820
- #9533

**Note:** I will do a hotfix in prod for the remove account case, because it's important to fix quickly (GDPR).

### CODE REVIEW

**Comment by alexgarel** [comment]
This is a big PR ! I still have to double read my changes to be sure.

File: templates/api/knowledge-panels/contribution/data_quality_tags.tt.json [hunk 1]
```diff
@@ -58,7 +58,7 @@
                 {
                     "element_type": "action",
                     "action_element": {
-                        "html": `[% lang(action_name) %]`,
+                        "html": `[% edq(lang(action_name)) %]`,
```
Comment (alexgarel) [inline templates/api/knowledge-panels/contribution/data_quality_tags.tt.json:61]: it's not needed here… as we are using back-thick.

I have to check in other places.

**Comment by alexgarel** [comment]
Ideally we should add a check that we don't use lang / f_lang without edq or esq if we are inside quotes. (but I don't think of an easy way to do this).

**Comment by alexgarel** [comment]
This is ready for review & merge @stephanegigandet

File: lib/ProductOpener/Display.pm [hunk 2]
```diff
@@ -5736,12 +5740,21 @@ sub search_and_export_products ($request_ref, $query_ref, $sort_by) {
 	return;
 }
 
+sub escape_char($s, $char) {
```
Comment (stephanegigandet) [inline lib/ProductOpener/Display.pm:5743]: Looks good. Maybe this function (and escape_single_quote) could be moved to Text.pm

**Review by stephanegigandet — APPROVED** [review]
Looks good, thank you!

**Pull request merged**